swap_meet/vendor.py

```diff
@@ -1,2 +1,120 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    # Creator: Each Vendor object has an attribute inventory
+    # The inventory should be empty if the caller does not
+    # provide any argument.
+    def __init__(self, inventory=None):
+        # If the caller did not provide an inventory,
+        # then create a new empty list and save in the new
+        # Vendor's object inventory
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    # method: add item to inventory
+    # the argument item is a reference to an object of class Item
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    # method: remove item from inventory
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    # test_wave_02
+    # method: create a list of item by category
+    def get_by_category(self, category):
+        list_item_by_category= []
+        for item in self.inventory:
+            if item.category == category:
+                list_item_by_category.append(item)
+        return list_item_by_category
+
+    # test_wave_03
+    # if my_item not in self_inventory and their_item not in vendor_inventory
+    # do not swap
+    def swap_items(self,vendor, my_item,their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        self.remove(my_item)
+        vendor.add(my_item)
+        vendor.remove(their_item)
+        self.add(their_item)
+        return True
+    #test-wave_04
+    def swap_first_item(self, vendor):
+        # removes the first item in self.inventory and add to vendor inventory
+        # removes the first item from friend inventory and add to self inventory
+        # return True
+        # if either self or friend has empty inventory, return False
+        
+        if not self.inventory or not vendor.inventory:
+            return False
+        first_item_in_self = self.inventory[0]
+        first_item_in_vendor = vendor.inventory[0]
+        self.remove(first_item_in_self)
+        vendor.remove(first_item_in_vendor)
+        vendor.add(first_item_in_self)
+        self.add(first_item_in_vendor)
```

Comment:
Hmm where have we seen this before? How can we use `swap_items` here instead of duplicating code?